### DISCUSSION TITLE
I now have a need to deduplicate vectors.

### DISCUSSION THREAD

**Discussion by luliloveshe** [milvus-io/milvus#27759] [Q&A and General discussion]
I now have a need to deduplicate vectors. After the audio search, 40 result audios were returned. Each audio is a set of vectors, and the number is more or less depending on the duration, such as a short one of 10x128, a long one of 30x128, and a longer one of 100x128. Please tell me how to remove duplicates from 40 audio files. For example, 3 audios are duplicated and 2 are duplicated. After deduplication, the number is 40-2-1=37.

**Comment by yhmo** [comment]
Then you need to define the rule of "duplicated".
I am not sure which way is best, I just give my proposal.
Assume we have two songs: A and B
A is converted to 10x128 vectors
B is converted to 30x128 vectors
We pick vectors of A one by one, to calculate the distance with each vector of B. If the distance value exceeds a threshold, then we say the two vectors are duplicated. If each vector of A has a duplicated vector in B, then we say A and B are duplicated.

**Reply by luliloveshe** [reply]
@yhmo 
I understand the way you put it, but the calculation scale may be too large. Suppose I have 40 search results, then according to the formula N * (N - 1) / 2, there are 40 * (40 - 1) / 2 = 780 pairwise calculations. The two sets of vectors calculated each time are used to calculate the distance, such as A:10x128, B:30x128. Is there a more efficient way?

**Comment by luliloveshe** [comment]
I have an idea but I don't know if it is feasible? I searched and found 40 results, all with scores. If several of the result scores are close to reaching a certain threshold, I consider them duplicates. This eliminates the need to calculate distances in pairs

**Reply by yhmo** [reply]
Hmmm... not sure what do you mean "searched and found 40 results".

**Reply by luliloveshe** [reply]
@yhmo 
I want to implement audio search and deduplication functions. The user uploads an audio, and the search returns 40 similar audios, and then these 40 audios are deduplicated.
